install-package

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+# Package installer with caching, version 2.2
+
+_show_usage_and_exit() {
+  echo -e "Package installer that caches packages and speeds up subsequent installations\n"
+  echo -e "Usage: install-package [--update] [--skip-update] [--update-new] pkg1[=version] [pkg2[=version]] [-o APT::Option] ...\n"
+  echo "--update            Forces repository list update before installing the packages"
+  echo "--skip-update       Skips repository list update"
+  echo "--update-new        Update only repository lists added in the last 1h"
+
+  exit 1
+}
+
+_skip_global_update() {
+  touch $apt_update_ran
+}
+
+_update_repo_lists() {
+  echo ">> Updating repository lists and keys..."
+
+  sudo apt-key update
+  sudo apt-get update
+
+  _skip_global_update
+}
+
+_install_packages() {
+  echo ">> Installing packages..."
+
+  sudo apt-get install -y --force-yes -o Dir::Cache::archives="$temp_deb_store" ${packages[@]}
+}
+
+_unpack_pkg_archive_if_available() {
+    echo ">> Unpacking package archive..."
+
+    cache restore install_package_cache
+
+    _skip_global_update
+}
+
+_archive_packages() {
```

Comment:
Why do we use `_` for function names?

Reply:
It was you or Ervin i suppose in the early days :), I just adjusted it to use the new caching system.

Reply:
I see. It was probably meant to indicate that this is `private` in the scope of the script.

There is no need for this IMHO anymore. Please remove the `_`.

Reply:
Done.